extensions/resteasy-reactive/rest-jackson/deployment/src/main/java/io/quarkus/resteasy/reactive/jackson/deployment/processor/JacksonCodeGenerator.java

```diff
@@ -0,0 +1,373 @@
+
+        private String jsonName() {
+            AnnotationInstance jsonProperty = annotations.get(JsonProperty.class.getName());
+            if (jsonProperty != null) {
+                AnnotationValue value = jsonProperty.value();
+                if (value != null && !value.asString().isEmpty()) {
+                    return value.asString();
+                }
+            }
+            return fieldName();
+        }
+
+        private String fieldName() {
+            return fieldInfo != null ? fieldInfo.name() : fieldNameFromMethod(methodInfo);
+        }
+
+        private String fieldNameFromMethod(MethodInfo methodInfo) {
+            String methodName = methodInfo.name();
+            if (methodName.startsWith("is")) {
+                return methodName.substring(2, 3).toLowerCase() + methodName.substring(3);
+            }
+            if (methodName.startsWith("get") || methodName.startsWith("set")) {
+                return methodName.substring(3, 4).toLowerCase() + methodName.substring(4);
+            }
+            return methodName;
+        }
+
+        boolean hasUnknownAnnotation() {
+            return annotations.keySet().stream()
+                    .anyMatch(ann -> ann.startsWith("com.fasterxml.jackson.") && !ann.equals(JsonProperty.class.getName()));
+        }
+
+        ResultHandle toValueWriterHandle(BytecodeCreator bytecode, ResultHandle valueHandle) {
+            return switch (fieldType.name().toString()) {
+                case "char", "java.lang.Character" -> bytecode.invokeVirtualMethod(
+                        MethodDescriptor.ofMethod(String.class, "charAt", char.class, int.class), valueHandle,
+                        bytecode.load(0));
+                default -> valueHandle;
+            };
+        }
+
+        ResultHandle toValueReaderHandle(BytecodeCreator bytecode, ResultHandle valueHandle) {
+            ResultHandle handle = accessorHandle(bytecode, valueHandle);
+
+            return switch (fieldType.name().toString()) {
+                case "char", "java.lang.Character" -> bytecode.invokeStaticMethod(
+                        MethodDescriptor.ofMethod(Character.class, "toString", String.class, char.class), handle);
+                default -> handle;
+            };
+        }
+
+        private ResultHandle accessorHandle(BytecodeCreator bytecode, ResultHandle valueHandle) {
+            if (methodInfo != null) {
+                if (methodInfo.declaringClass().isInterface()) {
+                    return bytecode.invokeInterfaceMethod(MethodDescriptor.of(methodInfo), valueHandle);
+                }
+                return bytecode.invokeVirtualMethod(MethodDescriptor.of(methodInfo), valueHandle);
+            }
+            return bytecode.readInstanceField(FieldDescriptor.of(fieldInfo), valueHandle);
+        }
+
+        String writtenType() {
+            return switch (fieldType.name().toString()) {
+                case "char", "java.lang.Character" -> "java.lang.String";
```

Comment:
This isn't supposed to map both to primitive char?

Reply:
No. Maybe the name of the method is misleading, anyway this is the type that has to be used when a field is written into json. In Jackson (and indeed in json) there's no notion of a single char, so it has to be written as a String.